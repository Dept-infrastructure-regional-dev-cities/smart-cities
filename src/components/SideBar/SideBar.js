import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import CityNavContainer from '../CityNavContainer/CityNavContainer';
import style from './SideBar.scss';

class SideBar extends Component {
  componentWillReceiveProps(nextProps) {
    // If side nav is open and the pathname just changed, that means we should close the nav
    if (nextProps.isOpen && this.props.location.pathname !== nextProps.location.pathname) {
      this.props.closeNav();
    }
  }

  render() {
    const { props } = this;

    const className = classNames(
      style.wrapper,
      {
        [style.showOpen]: props.isOpen,
      },
    );

    return (
      <div className={className}>
        <button
          className={style.button}
          onClick={props.closeNav}
          aria-label="close navigation menu"
        >âœ•</button>

        <h1>SideBar</h1>

        <CityNavContainer
          categoryId={props.categoryId}
          // pass in cityId so the nav updates when the city changes
          cityId={props.cityId}
        />
      </div>
    );
  }
}

SideBar.propTypes = {
  closeNav: PropTypes.func.isRequired,
  isOpen: PropTypes.bool.isRequired,
  location: PropTypes.shape({
    pathname: PropTypes.string,
  }),
  categoryId: PropTypes.string.isRequired,
  cityId: PropTypes.string.isRequired,
};

SideBar.defaultProps = {
  location: {},
};

export default SideBar;
