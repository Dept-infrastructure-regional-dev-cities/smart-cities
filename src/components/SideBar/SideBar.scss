@import '~style/variables';

.wrapper {
  position: fixed;
  top: 0;
  left: 0;
  width: $sidebar-width;
  bottom: 0;
  background: $color-primary--600;
}

.showOpen {
  transform: translateX(0);
}
