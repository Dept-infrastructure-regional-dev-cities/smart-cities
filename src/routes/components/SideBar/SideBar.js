import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import style from './SideBar.scss';

class SideBar extends Component {
  componentWillReceiveProps(nextProps) {
    // If side nav is open and the pathname just changed, that means we should close the nav
    if (nextProps.isOpen && this.props.location.pathname !== nextProps.location.pathname) {
      this.props.closeNav();
    }
  }

  render() {
    const className = classNames(
      style.wrapper,
      {
        [style.showOpen]: this.props.isOpen
      },
    );

    const {closeNav} = this.props;

    return (
      <div className={className}>
        <button
          className={style.button}
          onClick={closeNav}
          aria-label="close navigation menu"
        >âœ•</button>

        <h1>SideBar</h1>
      </div>
    );
  }
}

SideBar.propTypes = {
  closeNav: PropTypes.func.isRequired,
  isOpen: PropTypes.bool.isRequired,
};

export default SideBar;
