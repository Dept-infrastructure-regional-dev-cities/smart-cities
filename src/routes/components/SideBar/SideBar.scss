@import '~style/variables';

.wrapper {
  border: 1px solid blue;
  position: fixed;
  top: 0;
  left: 0;
  width: $sidebar_width;
  bottom: 0;
  background: $color-primary--400;
}

.showOpen {
  transform: translateX(0);
}

.button {
  color: red;
}
