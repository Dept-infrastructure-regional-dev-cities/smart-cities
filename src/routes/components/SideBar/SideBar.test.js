import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import SideBar from './SideBar';

const closeNavMock = jest.fn();

const defaultProps = {
  closeNav: closeNavMock,
  isOpen: false,
};

it('should match Snapshot', () => {
  const component = renderer.create(
    <SideBar {...defaultProps} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

it('should apply a class to open the component', () => {
  const component = shallow(
    <SideBar
      {...defaultProps}
      isOpen={true}
    />
  );

  const sideBar = component.find('.SideBar');

  expect(sideBar.hasClass('SideBar_open')).toBe(true);
});

it('should close the nav when the button is clicked', () => {
  const component = shallow(
    <SideBar {...defaultProps} />
  );

  component.find('.close_button').simulate('click');

  expect(closeNavMock).toHaveBeenCalledTimes(1);
});
