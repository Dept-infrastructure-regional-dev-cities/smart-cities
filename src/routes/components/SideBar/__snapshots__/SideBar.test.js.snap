// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
<div
  className="SideBar"
>
  <button
    aria-label="close navigation menu"
    className="sideNav_close_button"
    onClick={[Function]}
  >
    âœ•
  </button>
  <h1>
    SideBar
  </h1>
</div>
`;
